include/bout/physicsmodel.hxx

```diff
@@ -260,6 +271,10 @@ protected:
    * 
    */ 
   bool bout_constrain(Field3D &var, Field3D &F_var, const char *name);
+
+  // write restarts and pass outputMonitor method inside a Monitor subclass
+  friend class PhysicsModelMonitor;
```

Comment:
@ZedThree Could you explain for me why it's better to make PhysicsModelMonitor not be a friend? The reason I'm thinking that maybe this is a place to use `friend` is that looking at PhysicsModel, I saw that `timestepMonitor` is protected and called through `runTimestepMonitor` instead of just being public.
https://github.com/boutproject/BOUT-dev/blob/d6a89916ede53943810ff35312b68bc872026331/include/bout/physicsmodel.hxx#L166
I guess the reason for this interface is to allow the base `PhysicsModel` class to add things to `runTimestepMonitor` in future without changes to the API. That made me think, isn't a `friend` `PhysicsModelMonitor` doing a similar thing, providing a public API to protected parts of PhysicsModel? In this view, the PhysicsModelMonitor is 'part of' PhysicsModel that just happens to need to be a class itself.

Reply:
3af71fc: We can make PhysicsModelMonitor a nested class
https://en.cppreference.com/w/cpp/language/nested_types
I'd never heard of them before, but it avoids the need to make it a friend or to expose more members/methods of PhysicsModel.